### PR TITLE
Fenix #14344: Remove runBlocking in saved login dialog

### PR DESCRIPTION
**Pull request by eliserichards**
---

For Fenix https://github.com/mozilla-mobile/fenix/issues/14544

### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/master/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Milestone**: Make sure issues closed by this pull request are added to the [milestone](https://github.com/mozilla-mobile/android-components/milestones) of the version currently in development.
- **Breaking Changes**: If this is a breaking change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### CODE REVIEW

**Comment by eliserichards** [comment]
bors try

File: components/feature/prompts/src/main/java/mozilla/components/feature/prompts/dialog/SaveLoginDialogFragment.kt [hunk 1]
```diff
@@ -301,12 +300,17 @@ internal class SaveLoginDialogFragment : PromptDialogFragment() {
...
-        // the previous call has been canceled
-        runBlocking { validateStateUpdate?.cancelAndJoin() }
-        var validateDeferred: Deferred<Result>? = null
-        validateStateUpdate = scope.launch(IO) {
-            val validationDelegate = feature?.loginValidationDelegate ?: return@launch
+
+        try {
+            validateStateUpdate?.cancelAndJoin()
+        } catch (cancellationException: CancellationException) {
+            Logger.error("Failed to cancel save login dialog", cancellationException)
```
Comment (ekager) [inline components/feature/prompts/src/main/java/mozilla/components/feature/prompts/dialog/SaveLoginDialogFragment.kt:307]: nit - we're not cancelling the dialog, we're cancelling the validateStateUpdate job

File: components/feature/prompts/src/main/java/mozilla/components/feature/prompts/dialog/SaveLoginDialogFragment.kt [hunk 2]
```diff
@@ -301,12 +300,17 @@ internal class SaveLoginDialogFragment : PromptDialogFragment() {
...
+        }
+
+        var validateDeferred: Deferred<Result>?
```
Comment (ekager) [inline components/feature/prompts/src/main/java/mozilla/components/feature/prompts/dialog/SaveLoginDialogFragment.kt:310]: Any reason that we removed resetting this value to null?

File: components/feature/prompts/src/main/java/mozilla/components/feature/prompts/dialog/SaveLoginDialogFragment.kt (hunk 1 above)
Comment (ekager) [reply components/feature/prompts/src/main/java/mozilla/components/feature/prompts/dialog/SaveLoginDialogFragment.kt:307]: I'm don't even think we need/want to catch this. Looking at the docs for `CancellationException` - 
 * Thrown by cancellable suspending functions if the [Job] of the coroutine is cancelled while it is suspending.
 * It indicates _normal_ cancellation of a coroutine.
 * **It is not printed to console/log by default uncaught exception handler**.

**Comment by eliserichards** [comment]
Just tested it locally with Fenix. Looks like everything is working @ekager

**Review by ekager — APPROVED** [review]
👍

**Comment by eliserichards** [comment]
bors r=ekager

**Pull request merged**